File: graphql-test/src/main/java/com/junyss/graphqltest/api/team/model/entity/Team.java
package com.junyss.graphqltest.api.team.model.entity;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Entity
public class Team {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long teamId;
	private String manager;
	private String office;
	private String extensionNumber;
	private String mascot;
	private String cleaningDuty;
	private String project;

	@Builder
	public static Team toEntity(
		String manager,
		String office,
		String extensionNumber,
		String mascot,
		String cleaningDuty,
		String project
	) {
		return new Team(
			null,
			manager,
			office,
			extensionNumber,
			mascot,
			cleaningDuty,
			project);
	}

	public void updateManager(String manager) {
		this.manager = manager;
	}

	public void updateOffice(String office) {
		this.office = office;
	}

	public void updateExtensionNumber(String extensionNumber) {
		this.extensionNumber = extensionNumber;
	}

	public void updateMascot(String mascot) {
		this.mascot = mascot;
	}

	public void updateCleaningDuty(String cleaningDuty) {
		this.cleaningDuty = cleaningDuty;
	}

	public void updateProject(String project) {
		this.project = project;
	}
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/api/team/model/entity/Team.kt
package com.junyharangstudy.kotlingraphqltest.api.team.model.entity

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id

@Entity
class Team(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val teamId: Long?,
    var manager: String,
    var office: String,
    var mascot: String,
    var extensionNumber: String,
    var cleaningDuty: String,
    var project: String
) {
    fun updateManager(manager: String) {
        this.manager = manager
    }

    fun updateOffice(office: String) {
        this.office = office
    }

    fun updateMascot(mascot: String) {
        this.mascot = mascot
    }

    fun updateExtensionNumber(extensionNumber: String) {
        this.extensionNumber = extensionNumber
    }

    fun updateCleaningDuty(cleaningDuty: String) {
        this.cleaningDuty = cleaningDuty
    }

    fun updateProject(project: String) {
        this.project = project
    }
}